Src/FluentAssertions/Numeric/ComparableTypeAssertions.cs

```diff
@@ -49,7 +49,7 @@ public ComparableTypeAssertions(IComparable<T> value)
         public AndConstraint<TAssertions> Be(T expected, string because = "", params object[] becauseArgs)
         {
             Execute.Assertion
-                .ForCondition(Subject.IsSameOrEqualTo(expected))
+                .ForCondition(Equals(Subject, expected))
```

Comment:
I recall using `IsSameOrEqualTo` to make sure certain numeric values are still treated equally, even after boxing. Is this no longer needed? See https://github.com/fluentassertions/fluentassertions/blame/6170e2449326aeef60471ed4553b6ed416f9b51d/Src/FluentAssertions/Common/ObjectExtensions.cs#L33

Reply:
In this case Subject is of type `IComparer<T>` and `object` does not implement `IComparer<T>`.
One could have a subject of type `X : IComparer<Y>` so we cannot reliable use `EqualityComparer<T>.Default.Equals` here.

`IsSameOrEqualTo` is mostly relevant for `ObjectAssertions<object>.Be(TSubject expected)` and as https://github.com/fluentassertions/fluentassertions/commit/3fa634fd41f68ab19eeb89424e766b8e653e0212#diff-74519b899a32f7b49a7eaaff37fd152b3537fd970ee1ced4bcef91dd4e42cb37 shows this should mostly happen when comparing non-generic collections for equivalency.